app/MMB/src/navigation/TravelDocumentModalScreen.js

```diff
@@ -63,24 +76,70 @@ export default class TravelDocumentModalScreen extends React.Component<
         </HeaderTitle>
       ),
       headerRight: (
-        <HeaderButton onPress={goBack}>
+        <HeaderButton
+          onPress={onSave}
+          disabled={navigation.state.params.disabled}
+        >
           <Translation id="mmb.missing_informaiton.travel_document_modal_screen.save" />
         </HeaderButton>
       ),
       gesturesEnabled: false,
     };
   };
 
+  componentDidMount = () => {
+    this.props.navigation.setParams({ onSave: this.onSave, disabled: true });
+  };
+
   onIdNumberChange = (idNumber: string) => {
-    this.setState({ idNumber });
+    this.setState(
+      state => ({
+        idNumber,
+        error: {
+          ...state.error,
+          idNumber: idNumber.length < 5 && idNumber.length !== 0,
+        },
+      }),
+      this.validate,
+    );
   };
 
   onDateChange = (expiryDate: Date) => {
-    this.setState({ expiryDate });
+    this.setState(
+      state => ({
+        expiryDate,
+        error: {
+          ...state.error,
+          expiryDate: false,
+        },
+      }),
+      this.validate,
+    );
   };
 
   onNoExpiryChange = (noExpiry: boolean) => {
-    this.setState({ noExpiry });
+    this.setState(
+      state => ({
+        noExpiry,
+        expiryDate: null,
+        error: {
+          ...state.error,
+          expiryDate: !noExpiry,
+        },
+      }),
+      this.validate,
+    );
+  };
+
+  onSave = () => {
+    // TODO: Send mutation
+    this.props.navigation.goBack();
+  };
+
+  validate = () => {
+    const hasError =
+      this.state.idNumber.length < 5 || this.state.error.expiryDate;
```

Comment:
And here you are using a different conditional for the idNumber

Reply:
You probably wanted `this.state.error.idNumber`?

Reply:
Actually not, but it should be commented. If idNumber.length === 0. I do not want to show error message below the input, but I want to disable header button

Reply:
That makes sense. 👍